src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs

```diff
@@ -77,5 +78,113 @@ public void DeletePolicy()
+            DateTime retentionTime = GenerateRandomTime(); 
+
+            //Daily Retention policy
+            defaultRetention.IsDailyScheduleEnabled = true;
+            defaultRetention.DailySchedule = new Models.DailyRetentionSchedule();
+            defaultRetention.DailySchedule.RetentionTimes = new List<DateTime>();
+            defaultRetention.DailySchedule.RetentionTimes.Add(retentionTime);
+            defaultRetention.DailySchedule.DurationCountInDays = 180; //TBD make it const
+
+            //Weekly Retention policy
+            defaultRetention.IsWeeklyScheduleEnabled = true;
+            defaultRetention.WeeklySchedule = new Models.WeeklyRetentionSchedule();
+            defaultRetention.WeeklySchedule.DaysOfTheWeek = new List<DayOfWeek>();
+            defaultRetention.WeeklySchedule.DaysOfTheWeek.Add(DayOfWeek.Sunday);
+            defaultRetention.WeeklySchedule.DurationCountInWeeks = 104; //TBD make it const
+            defaultRetention.WeeklySchedule.RetentionTimes = new List<DateTime>();
+            defaultRetention.WeeklySchedule.RetentionTimes.Add(retentionTime);
+
+            //Monthly retention policy
+            defaultRetention.IsMonthlyScheduleEnabled = true;
+            defaultRetention.MonthlySchedule = new Models.MonthlyRetentionSchedule();
+            defaultRetention.MonthlySchedule.DurationCountInMonths = 60; //tbd: make it const
+            defaultRetention.MonthlySchedule.RetentionScheduleFormatType = Models.RetentionScheduleFormat.Weekly;
+
+            //Initialize day based schedule
+            defaultRetention.MonthlySchedule.RetentionScheduleDaily = GetDailyRetentionFormat();  
+
+            //Initialize Week based schedule
+            defaultRetention.MonthlySchedule.RetentionScheduleWeekly = GetWeeklyRetentionFormat();
+
+            //Yearly retention policy
+            defaultRetention.IsYearlyScheduleEnabled = true;
+            defaultRetention.YearlySchedule = new Models.YearlyRetentionSchedule();
+            defaultRetention.YearlySchedule.DurationCountInYears = 10;
+            defaultRetention.YearlySchedule.RetentionScheduleFormatType = Models.RetentionScheduleFormat.Weekly;
+            defaultRetention.YearlySchedule.MonthsOfYear = new List<Models.Month>();
+            defaultRetention.YearlySchedule.MonthsOfYear.Add(Models.Month.January);
+            defaultRetention.YearlySchedule.RetentionScheduleDaily = GetDailyRetentionFormat();
+            defaultRetention.YearlySchedule.RetentionScheduleWeekly = GetWeeklyRetentionFormat();
+            return defaultRetention;
+
+        }
+
+        private static Models.DailyRetentionFormat GetDailyRetentionFormat()
+        {
+            Models.DailyRetentionFormat dailyRetention = new Models.DailyRetentionFormat();
+            dailyRetention.DaysOfTheMonth = new List<Models.Day>();
+            Models.Day dayBasedRetention = new Models.Day();
+            dayBasedRetention.IsLast = false;
+            dayBasedRetention.Date = 1;
```

Comment:
Please check default values with portal and put same values everywhere here like date=1, day=sunday, month-January etc